processing/src/main/java/io/druid/query/topn/AlphaNumericTopNMetricSpec.java

```diff
@@ -38,7 +38,11 @@ public int compare(String str1, String str2)
     {
       int[] pos = {0, 0};
 
-      if (str1.length() == 0) {
+      if (str1 == null) {
+        return str2 == null ? 0 : -1;
```

Comment:
can you just return -1 here, so that it will be more predictable and and consistent with LexicographicTopNMetricSpec ?